components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementAuditLogger.java

```diff
@@ -223,7 +226,15 @@ public boolean doPostUpdateCredentialByAdmin(String userName, Object newCredenti
             UserStoreManager userStoreManager) {
 
         if (isEnable()) {
-            audit.info(createAuditMessage(ListenerUtils.CHANGE_PASSWORD_BY_ADMIN_ACTION, getTargetForAuditLog
+            IdentityContext identityContext = IdentityContext.getThreadLocalIdentityContext();
```

Comment:
Let's get the flow directly from here. 
```suggestion
            Flow flow = IdentityContext.getThreadLocalIdentityContext().getFlow();
```
Let's get the flow directly. And we need to check whether the flow is notnull. Otherwise below line can throw a NPE.

Check: https://github.com/wso2/carbon-identity-framework/blob/712662e074f569aa2c969e2f55cdba9a87a60ed9/components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/internal/execution/PreUpdatePasswordRequestBuilder.java#L114